nixos/modules/services/networking/nomad.nix

```diff
@@ -34,6 +34,18 @@ in
           Configuration for Nomad. See the <link xlink:href="https://www.nomadproject.io/docs/configuration">documentation</link>
           for supported values.
         '';
+        example = literalExample ''
+          # A minimal config example:
+          data_dir = "/var/lib/''${config.systemd.services.nomad.serviceConfig.StateDirectory}";
```

Comment:
I think there's a slight chicken-egg problem here. If user configures `data_dir = "/my/nomad/data";` we'd still create `StateDirectory` not at that location. What should we do with that? Only set `StateDirectory` if the dirname of `data_dir` is `/var/lib`?

Reply:
My preferred way is to only provision the data/state directory if the user hasn't "customized" it. In [this example](https://github.com/NixOS/nixpkgs/pull/106697) there is a `dataDir` option, but the same idea can apply here:
```
serviceConfig = mkMerge [
  { # standard options
  }
  (mkIf (cfg.settings.data_dir == "/var/lib/nomad") { StateDirectory = "nomad"; })
];
```

I take special care to explain to the sysadmin their responsibilities, if any:
> The directory or NFS/SMB network share where MPD reads music from. If left
as the default value this directory will automatically be created before
the MPD server starts, otherwise the sysadmin is responsible for ensuring
the directory exists with appropriate ownership and permissions.

Why? See the [motivation section](https://github.com/NixOS/nixpkgs/pull/106697#issue-537600068) of the mentioned PR.